### PR TITLE
Assert migration index at startup

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Detect misconfigured setups and prevent wrong data usage.

### CODE REVIEW

File: dev/Dockerfile.dev [hunk 1]
```diff
@@ -19,7 +19,7 @@ COPY Makefile Makefile
 COPY setup.cfg .
 COPY dev/entrypoint.sh ./
 
-COPY migrations/migrate.py migrations/.
+COPY migrations/*.py migrations/
 COPY migrations/migrations migrations/migrations/.
```
Comment (jsangmeister) [inline dev/Dockerfile.dev:23]: Wouldn't it be better to mount these folders?

File: openslides_backend/models/assert_migration_index.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from datastore.shared.di import injector
```
Comment (jsangmeister) [inline openslides_backend/models/assert_migration_index.py:1]: I don't think this belongs in the `models` folder, but rather in the `migrations` folder

File: openslides_backend/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .assert_migration_index import assert_migration_index  # noqa
```
Comment (jsangmeister) [inline openslides_backend/models/__init__.py:1]: In the backend, we do not use the `__init__.py` for such exports, so I don't think it's fitting to use it here in this way

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: dev/Dockerfile.dev (hunk 1 above)
Comment (FinnStutzenstein) [reply dev/Dockerfile.dev:23]: I thought about it, but when *developing* migrations, there is a dedicated setup in the migrations folder. Until you want to change the intersection (load_modules), you have to rebuild/test-edit with vim. I had to rebuild 3 times, which is a low quote I guess to make it work. I think there is currently no need

File: openslides_backend/models/assert_migration_index.py (hunk 2 above)
Comment (FinnStutzenstein) [reply openslides_backend/models/assert_migration_index.py:1]: ok, I'll move it.

File: dev/Dockerfile.dev (hunk 1 above)
Comment (jsangmeister) [reply dev/Dockerfile.dev:23]: I found that mounting is almost always better than copying because it eliminates one source of programming mistakes and just makes the experience a lot smoother if you need it. But I see your point

**Comment by FinnStutzenstein** [comment]
@jsangmeister The action "Build and test production docker image" does not work as is:
```
psycopg2.OperationalError: could not translate host name "postgresql" to address: No address associated with hostname
```

Which is sensible since on startup a DB query is made. I can either remove this action or use a docker-compose setup. Whats your opinion?

**Comment by jsangmeister** [comment]
> @jsangmeister The action "Build and test production docker image" does not work as is:
> 
> ```
> psycopg2.OperationalError: could not translate host name "postgresql" to address: No address associated with hostname
> ```
> 
> Which is sensible since on startup a DB query is made. I can either remove this action or use a docker-compose setup. Whats your opinion?

I would prefer the latter, it's better to keep tests if possible

File: dev/Dockerfile.dev (hunk 1 above)
Comment (FinnStutzenstein) [reply dev/Dockerfile.dev]: I mounted it now ;)

File: dev/Dockerfile.dev [hunk 4]
```diff
@@ -10,7 +10,7 @@ COPY dev/requirements_development.txt dev/.
 
 RUN pip install --no-cache-dir --requirement dev/requirements_development.txt
 
-COPY dev/wait.sh dev/.
+COPY wait.sh ./
```
Comment (jsangmeister) [inline dev/Dockerfile.dev:13]: Why did you move the file? I think it belongs in `/dev/`

Comment (FinnStutzenstein) [reply dev/Dockerfile.dev:13]: It is not used in the main Dockerfile

**Review by jsangmeister — APPROVED** [review]
